### PR TITLE
Write an error for powershell -version <any value>

### LINKED ISSUE #4834: powershell -version 2 should return error
As far as I know, PowerShell v2 was recently declared deprecated.
Thus, it normal to expect some kind of related errors in the latest v6.

The behaviour is strange, though. If I try to invoke something (anything
actually, including gibberish) with `-version 2` then PowerShell succeeds
(`$LASTEXITCODE` is 0) and prints the same result `powershell v6.0.0-beta.7`.

Steps to reproduce
------------------

````powershell
    powershell -version 2 gibberish
    $LASTEXITCODE
````

Expected behavior
-----------------

Non-zero `$LASTEXITCODE` and a clear error message.

Actual behavior
---------------

Zero `$LASTEXITCODE` and some version-like text, the same regardless of a command.

````
    powershell v6.0.0-beta.7
    0
````

Environment data
----------------

PowerShell-6.0.0-beta.7-win-x64.zip

````
Name                           Value
----                           -----
PSVersion                      6.0.0-beta
PSEdition                      Core
GitCommitId                    v6.0.0-beta.7
OS                             Microsoft Windows 10.0.10586
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
````

### PR DESCRIPTION
**Pull request by iSazonov**
Fix #4834

### CODE REVIEW

File: src/Microsoft.PowerShell.ConsoleHost/resources/CommandLineParameterParserStrings.resx [hunk 1]
```diff
@@ -272,4 +272,7 @@ EXAMPLES
   <data name="InvalidArgument" xml:space="preserve">
     <value>Invalid argument '{0}', did you mean:</value>
   </data>
+  <data name="DeprecatedVersionParameter" xml:space="preserve">
+    <value>Old functionality of '-version 2.0' is deprecated. Now '-version' parameter shows current PowerShell version. </value>
```
Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.ConsoleHost/resources/CommandLineParameterParserStrings.resx]: I think we should change this error message to:

```none
Previous usage of '-version {0}' is not supported.  '-version' currently only supports returning the current PowerShell version.
```

cc @joeyaiello

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs [hunk 2]
```diff
@@ -548,6 +548,15 @@ private void ParseHelper(string[] args)
                     _noInteractive = true;
                     _skipUserInit = true;
                     _noExit = false;
+
+                    ++i;
+                    if (i < args.Length)
+                    {
+                        if (LanguagePrimitives.TryConvertTo<int>(args[i], out int verNumber) && verNumber == 2)
```
Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs]: I think we should currently error if any value is passed to `-v`

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs (hunk 2 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs]: Now we write the error and then show current version.
Please clarify your request.

Comment (SteveL-MSFT) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs]: Aren't you only writing error if `-v 2`?  Should also error if `-v 3`

Comment (iSazonov) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs]: I agree. Will add the error for `-v 3` too.

Comment (SteveL-MSFT) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs]: We should error on any value passed to `-v`.  In some future time if we decide to support launching different installed versions of PSCore6, we can decide if we want to overload `-v` or even introduce a new parameter.

Comment (iSazonov) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs]: Currently we ignore all values after `-v`. The docs says the same: accept only 2.0 or 3.0. If anybody use the parameter the breaking change is related only to 2.0 or 3.0 values - we should write an error for the cases only.

File: src/Microsoft.PowerShell.ConsoleHost/resources/CommandLineParameterParserStrings.resx (hunk 1 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.ConsoleHost/resources/CommandLineParameterParserStrings.resx]: I changed the error message - please review.

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs (hunk 2 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs]: Could you comment #4836 - I am ready to implement enumeration of installed versioms.

Comment (anmenaga) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs]: If we currently ignore all values after -v, then it seems logical to write error on any value passed to -v saying that parameter doesn't have any effect.
Having a parameter that quietly does nothing doesn't look good.

**Review by anmenaga — CHANGES_REQUESTED** [review]
Please add tests.

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs [hunk 3]
```diff
@@ -548,6 +548,15 @@ private void ParseHelper(string[] args)
                     _noInteractive = true;
                     _skipUserInit = true;
                     _noExit = false;
+
+                    ++i;
+                    if (i < args.Length)
+                    {
+                        if (LanguagePrimitives.TryConvertTo<int>(args[i], out int verNumber) && (verNumber == 2 || verNumber == 3))
```
Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs]: I think we should error on any value passed to `-v` as part of this PR.

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs (hunk 2 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs]: It is an enhancement. It make sense. Thanks!

Done.

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs (hunk 3 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs]: Done.

**Review by SteveL-MSFT — APPROVED** [review]

**Review by anmenaga — APPROVED** [review]

**Pull request merged**